portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -89,23 +89,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
 
     int numQueries = getLimit(request);
-    List<Entity> limitedResults = results.asList(FetchOptions.Builder.withLimit(10));
-    if(numQueries != -1){
-        limitedResults = results.asList(FetchOptions.Builder.withLimit(numQueries));
+    int defaultLimit = 10; 
+  
+    if(numQueries < 0 || numQueries > 10){
+        numQuries = defaultLimit;
+        System.err.println("Player choice is out of range. Defaulting to 10 comments.");
     }
-    // results.asIterable()
+    List<Entity> limitedResults = results.asList(FetchOptions.Builder.withLimit(numQueries));
+  
     ArrayList<Comment> comments = new ArrayList<>();
     for (Entity entity : limitedResults) {
       Comment comment = entityToComment(entity);
       comments.add(comment);
     }
     Gson gson = new Gson();
-
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(comments));
  }
 
  private int getLimit(HttpServletRequest request){
+     //if an error happens, returns -1 
```

Comment:
comments that explain functionality of a method usually go above the declaration. Java has a really fun (read:tedious) feature called "javadoc" that requires function and class documentation to be formatted in. 
 a special way. I won't require you do it here but it might be good to read up on it. this is justt one link but there are many good resources for javadoc: 
https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html